### PR TITLE
feat: support network logs obfuscation

### PR DESCRIPTION
**Pull request by a7medev**
## Description of the change

Add network logs obfuscation support through the new `NetworkLogger.obfuscateLog` API.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
## Related issues
Jira ID: MOB-12523
### Development
- [ ] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
### Code review 
- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: lib/src/modules/network_logger.dart [hunk 1]
```diff
@@ -17,8 +19,21 @@ class NetworkLogger {
...
+  /// @nodoc
+  @visibleForTesting
+  // ignore: use_setters_to_change_properties
+  static void $setManager(NetworkManager manager) {
+    _manager = manager;
+  }
+
+  static void obfuscateLog(ObfuscateLogCallback callback) {
+    _manager.setObfuscateLogCallback(callback);
+  }
```
Comment (abdelhamid-f-nasser) [inline lib/src/modules/network_logger.dart:31]: I suggest that we add docs with example to this method, what do you think? (a doc with an example for usage)
```dart
/// Registers a callback to obfuscate sensitive network log data.
///
/// Use this method to define a callback that will modify or obscure
/// sensitive information within network logs before they are recorded
/// or displayed.
///
/// The [callback] function takes a [NetworkData] object and should
/// return a modified [NetworkData] object.
///
/// Example:
/// 
/// NetworkLogger.obfuscateLog((data) {
///   // Implement custom logic to obfuscate sensitive data here.
///   // Modify 'data' as needed and return it.
/// });
/// 
static void obfuscateLog(ObfuscateLogCallback callback) {
  _manager.setObfuscateLogCallback(callback);
}
```

**Review by abdelhamid-f-nasser — APPROVED** [review]
Great job @a7medev 👏 . The code is well written and quite descriptive. Just a suggestion for adding docs to `obfuscateLog` API.

**Review by abdelhamid-f-nasser — COMMENTED** [review]
Great job @a7medev 👏 . The code is well written and quite descriptive. Just a suggestion for adding docs to `obfuscateLog` API.

File: lib/src/modules/network_logger.dart (hunk 1 above)
Comment (abdelhamid-f-nasser) [reply lib/src/modules/network_logger.dart:31]: @a7medev, I've made the updated mentioned here with a slight adjustment, could you please check them?

Comment (a7medev) [reply lib/src/modules/network_logger.dart:44]: Thanks a lot, @abdelhamid-f-nasser, once again! I've made small changes and moved it to `NetworkLogger` as well 🙏🏼

**Review by abdelhamid-f-nasser — APPROVED** [review]

**Pull request merged**